be/src/runtime/routine_load/routine_load_task_executor.cpp

```diff
@@ -273,34 +275,33 @@ void RoutineLoadTaskExecutor::exec_task(StreamLoadContext* ctx, DataConsumerPool
     switch (ctx->load_src_type) {
     case TLoadSourceType::KAFKA: {
         std::shared_ptr<DataConsumer> consumer;
-        Status st = _data_consumer_pool.get_consumer(ctx, &consumer);
+        Status st = consumer_pool->get_consumer(ctx, &consumer);
```

Comment:
If the status is not OK, you may not get the consumer from the pool. Then the return_consumer is wrong.